types/aws4/ts3.9/index.d.ts

```diff
@@ -0,0 +1,216 @@
+/// <reference types="node" />
+
+import { OutgoingHttpHeaders, RequestOptions } from 'http';
+
+export interface Request extends RequestOptions {
+    /** Defaults to {@link RequestSigner.createHost}() if possible. */
+    host?: string;
+    /** Equivalent to {@link host} */
+    hostname?: string;
+    /** Defaults to `"GET"`, or `"POST"` if there is a {@link body}. */
+    method?: string;
+    /** Defaults to `/`. */
+    path?: string;
+    /** Defaults to `""`. */
+    body?: string;
+    /** Defaults to {@link RequestSigner.matchHost}()[0], or `""`. */
+    service?: string;
+    /** Defaults to {@link RequestSigner.matchHost}()[1], or `"us-east-1"`. */
+    region?: string;
+    /** To sign the query instead of adding an Authorization header, defaults to false. */
+    signQuery?: boolean;
+    /**
+     * Some headers have a default value:
+     * - `Host`: Defaults to {@link hostname}, {@link host}, or
+     *   {@link RequestSigner.createHost}()
+     * - `Content-Type`: defaults to
+     *   `"application/x-www-form-urlencoded; charset=utf-8"`
+     *   if there is a {@link body}
+     * - `Date`: used to calculate the signature date if given, defaults to
+     * `new Date()`
+     */
+    headers?: OutgoingHttpHeaders;
+    /** If true, signing the Request won’t mutate the headers. */
+    doNotModifyHeaders?: boolean;
+    /** If true, {@link path} won’t be encoded in the signature. */
+    doNotEncodePath?: boolean;
+}
+
+export interface Credentials {
+    /** Equivalent to the env vars `AWS_ACCESS_KEY_ID` and `AWS_ACCESS_KEY`. */
+    accessKeyId?: string;
+    /** Equivalent to the env vars `AWS_SECRET_ACCESS_KEY` and `AWS_SECRET_KEY`. */
+    secretAccessKey?: string;
+    /** Equivalent to the env var `AWS_SESSION_TOKEN`. */
+    sessionToken?: string;
+}
+
+export class RequestSigner {
+    constructor(requestOptions: string | Request, credentials?: Credentials);
+
+    request: Request;
+    credentials: Credentials;
+    service: string;
+    region: string;
+
+    /**
+     * `true` if {@link service} is `"codecommit"` and {@link request}.method
+     * is `"GIT"`.
+     */
+    isCodeCommitGit: boolean;
+
+    /**
+     * Set by:
+     * - {@link prepareRequest}
+     * - {@link sign}
+     * - {@link canonicalString}
+     * - {@link parsePath}
+     */
+    parsedPath?: {
+        path: string;
+        query: Record<string, string | string[]>;
+    };
+
+    /**
+     * Used as a cache by {@link getDateTime} and {@link getDate}.
+     * Setting it will shortcut those functions.
+     *
+     * Set by:
+     * - {@link prepareRequest}
+     * - {@link getDateTime}
+     * - {@link sign}
+     */
+    datetime?: string;
+
+    /**
+     * Extract the service code and region code from a Host name.
+     */
+    matchHost(host: string): string[];
```

Comment:
This can still have a return type of `[string, string]`, right? It’s just the labels that are incompatible with 3.9 IIRC.

Reply:
You recall 100% correctly. I've added a commit updating this here and in the ts3.9 test.